src/error.c

```diff
@@ -140,7 +140,7 @@ static const char* find_last_newline(
     // There may be an error at EOF, which would be a nul byte.
     assert(*c || c == error_location);
   }
-  return c == original_text ? c : c + 1;
+  return c == original_text || c == error_location ? c : c + 1;
```

Comment:
Is this actually the correct fix? If `*error_location` is `\n`, shouldn't this move to the previous `\n` (if any)? For example, given the source text `"<\n"`, that line is the one with the error. With your patch, wouldn't `original_line` in `gumbo_caret_diagnostic_to_string()` point to the new line and have length 0?